plugins/woocommerce/src/Database/Migrations/CustomOrderTable/PostsToOrdersMigrationController.php

```diff
@@ -71,9 +72,101 @@ public function get_migrated_meta_keys() {
 	public function migrate_orders( array $order_post_ids ): void {
 		$this->error_logger = WC()->call_function( 'wc_get_logger' );
 
+		$using_transactions = $this->maybe_start_transaction();
+		if ( null === $using_transactions ) {
+			return;
+		}
+
+		$errors_were_logged = false;
+
 		foreach ( $this->all_migrators as $migrator ) {
-			$this->do_orders_migration_step( $migrator, $order_post_ids );
+			$errors_were_logged = $this->do_orders_migration_step( $migrator, $order_post_ids );
+			if ( $errors_were_logged && $using_transactions ) {
+				$this->rollback_transaction();
+				break;
+			}
+		}
+
+		if ( ! $errors_were_logged && $using_transactions ) {
+			$this->commit_transaction();
+		}
+	}
+
+	/**
+	 * Start a database transaction if the configuration mandates so.
+	 *
+	 * @return bool|null True if transaction started, false if transactions won't be used, null if transaction failed to start.
+	 */
+	private function maybe_start_transaction(): ?bool {
+		if ( 'yes' !== get_option( CustomOrdersTableController::USE_DB_TRANSACTIONS_OPTION ) ) {
+			return false;
+		}
+
+		$transaction_isolation_level             = get_option( CustomOrdersTableController::DB_TRANSACTIONS_ISOLATION_LEVEL_OPTION, CustomOrdersTableController::DEFAULT_DB_TRANSACTIONS_ISOLATION_LEVEL );
+		$set_transaction_isolation_level_command = "SET TRANSACTION ISOLATION LEVEL $transaction_isolation_level";
+
+		if ( ! $this->db_query( $set_transaction_isolation_level_command ) ) {
```

Comment:
The linter isn't flagging it because we're using our own method instead of `$wpdb->query()` directly, but otherwise this would trigger:

`Use placeholders and $wpdb->prepare(); found $set_transaction_isolation_level_command`

I guess in this case we can't pass the isolation level string through `$wpdb->prepare()`, but what about comparing it against a set of allowed values (or even escaping it via a preg_replace to strip anything but alphabetical and whitespace chars)?